runtime/common/bigint.go

```diff
@@ -54,5 +56,9 @@ func OverEstimateBigIntFromString(s string) int {
 	// First convert to word, and then find the size,
 	// to be consistent with `common.BigIntByteLength`
 	wordLen := (bitLen + 63) / 64
-	return (wordLen + 4) * BigIntWordSize
+	result := wordLen * BigIntWordSize
```

Comment:
We can make this check higher in the code :
```
func OverEstimateBigIntFromString(s string) int {
	l := len(s) 
	if l <= 2*BigIntWordSize   { // up to 16 hex characters fit into a 64-bits word
              return BigIntWordSize
         }
```
More details here: https://github.com/dapperlabs/cadence-private-issues/issues/41#issuecomment-1295666060

Reply:
@turbolent oh you're right, my bad I was assuming inputs are in base 16.

> Integer literals may have different bases (we support 2, 8, 10, 16)

The function implementation so far only considers base `10`. In general it would be great to cover other bases too, maybe 2 and 16.

Reply:
@tarakby  added the support for handling other bases too. Can you please check the equations above?

Also, how the ["test to check if the string fits in one word only"](https://github.com/onflow/cadence/pull/2110/files#r1008597788)" can be added now? I guess this is now needed to be handled separately for each base type?